### PR TITLE
FM2-136 : Appointment resource initial implementation

### PR DESCRIPTION
**Pull request by jecihjoy**
## Description of what I changed
Added an initial implementation for appointment  resource

## Issue I worked on

see https://issues.openmrs.org/browse/FM2-136

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**.

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/AppointmentTranslator.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.translators;
+
+import org.hl7.fhir.r4.model.Appointment;
+
+public interface AppointmentTranslator<T> extends ToFhirTranslator<T, Appointment>, UpdatableOpenmrsTranslator<T, Appointment> {
+	
+	/**
+	 * Maps Bahmni/OpenMRS appointment to {@link org.hl7.fhir.r4.model.Appointment}
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/translators/AppointmentTranslator.java:17]: I'd remove the reference to Bahmni from here. That's more of an implementation detail.

**Comment by mozzy11** [comment]
well done @jecihjoy  ,
will you implement other Operations like Update ,Search etc in a different PR ??

**Review by sherrif10 — APPROVED** [review]
This can however be merged to free this [PR ](https://issues.openmrs.org/browse/FM2-341), And it will leverage code workflow and tracking, Am thinking since this is the initial implementation it would be more clear to track different changes that may come along, cc @mozzy11 we also need to track mappings for appointmentResources FYI  cc @ibacher  @mozzy11

**Comment by corneliouzbett** [comment]
> This can however be merged to free this [PR ](https://issues.openmrs.org/browse/FM2-341), And it will leverage code workflow and tracking, Am thinking since this is the initial implementation it would be more clear to track different changes that may come along, cc @mozzy11 we also need to track mappings for appointmentResources FYI cc @ibacher @mozzy11

Hello @sherrif10 
There was a decision made in one of the FHIR squad meetings, suggesting this kind of implementation be implemented in the dependant module i.e this code should be moved to `bahmni appointments` module. Oh! forgot there's already a PR for this [here](https://github.com/Bahmni/openmrs-module-appointments/pull/29). Perhaps, we should be closing this PR.

**Pull request closed without merging**